### ISSUE TITLE
Instances of com.spotify.docker.client.messages.Container serialize to an empty JSON object when using Jackson ObjectMapper

### ISSUE THREAD

**Issue by RobertoGraham** [spotify/docker-client#1126]
**Is this a BUG REPORT or FEATURE REQUEST?**:

BUG REPORT

## Description

Instances of `com.spotify.docker.client.messages.Container` serialize to `{}`

## How to reproduce

Pass a `com.spotify.docker.client.messages.Container` instance to Jackson `ObjectMapper.writeValueAsString`. I have a demo Spring Boot application [here](https://github.com/RobertoGraham/docker-client-model-serializing). The demo application implicitly serializes a list of `com.spotify.docker.client.messages.Container` like this:

```java
@GetMapping(value = "/", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
public List<Container> containerList() throws DockerException, InterruptedException {
    return dockerClient.listContainers(
            allContainers()
    );
}
```

To use the demo application, make sure you have Docker running, make sure you have at least one container in any state, and then run the demo application from the entrypoint in  `src/main/java/com/example/demo/DemoApplication.java` and navigate to `localhost:8080`

## What do you expect

I expect a non-empty JSON object when serializing `com.spotify.docker.client.messages.Container` instance that looks similar to the output of `docker inspect <container ID>`

## What happened instead

Instead, `com.spotify.docker.client.messages.Container` instances serialize to `{}`

## Software:

- `docker version`: 
```text
Client: Docker Engine - Community
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        4d60db4
 Built:             Wed Nov  7 00:47:51 2018
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:55:00 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```
- Spotify's docker-client version: 8.14.5
## Full backtrace

```text
No backtrace sorry
```

**Comment by RobertoGraham** [comment]
I believe the Jackson annotations in `com.spotify.docker.client.messages.Container` aren't being picked up as docker-client is using a shaded Jackson. Do you guys have an example of making a non-shaded docker-client work?

**Comment by RobertoGraham** [comment]
Here's my workaround, I make use of your shaded Jackson dependencies when serializing models:

```java
import com.spotify.docker.client.shaded.com.fasterxml.jackson.databind.ObjectMapper;
import com.spotify.docker.client.shaded.com.fasterxml.jackson.datatype.guava.GuavaModule;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import static com.spotify.docker.client.shaded.com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;

@Configuration
public class ShadedObjectMapperConfiguration {

    @Bean
    public ObjectMapper shadedObjectMapper() {
        return new ObjectMapper()
                .registerModule(new GuavaModule())
                .setSerializationInclusion(NON_NULL);
    }
}
```

How come you have an extra Jackson module in `ObjectMapperProvider` that turns Sets into Maps?

**Comment by mattnworb** [comment]
Can you simply not depend on the shaded version of the docker-client JAR? e.g.

```xml
<dependency>
  <groupId>com.spotify</groupId>
  <artifactId>docker-client</artifactId>
  <version>8.14.5</version>
</dependency>

**Comment by RobertoGraham** [comment]
I do use the shaded JAR, @mattnworb. This is what's causing me the problem. It would be cleaner to use the non-shaded JAR so I could leave out that workaround but I don't which dependencies exactly are shaded and their versions, are they documented anywhere? I looked at `pom.xml` but just see package names

**Comment by mattnworb** [comment]
Right, that is what I was trying to suggest - using the non-shaded JAR. 

This section of the POM specifies which dependencies to include in the shaded JAR, and the `<relocations>` section of the POM that follows it details what packages to relocate:

https://github.com/spotify/docker-client/blob/6580ae844a3be96421dfce78707163526c8ba82c/pom.xml#L333-L345

**Comment by RobertoGraham** [comment]
Here are the changes required to make it work if anyone else encounters this: https://github.com/RobertoGraham/docker-client-model-serializing/commit/3a66b37536a63c35b2075dfc9e97d5e5760f7949. Thank you @mattnworb for your direction
